awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/AwsAppSignalsCustomizerProvider.java

```diff
@@ -79,7 +80,8 @@ private SdkTracerProviderBuilder customizeTracerProviderBuilder(
       logger.info("Span Metrics Processor enabled");
       String smpEndpoint =
           configProps.getString(
-              "otel.aws.smp.exporter.endpoint", "http://cloudwatch-agent.amazon-cloudwatch:4317");
+              "otel.aws.app.signals.exporter.endpoint",
+              configProps.getString("otel.aws.smp.exporter.endpoint", "http://localhost:4315"));
```

Comment:
Let me try to explain this in 2 parts:
1. The `cloudwatch-agent.amazon-cloudwatch` resolves to the CWAgent's K8s pod which is created by the CWOperator, [which will also set the "OTEL_AWS_SMP_EXPORTER_ENDPOINT"](https://github.com/aws/amazon-cloudwatch-agent-operator/blob/0bb89bd770fa2ff4fc74979110a11db8bfffbaa9/pkg/instrumentation/defaultinstrumentation.go#L32) environment variable. In this case, the default value will never be utilized. So, IMO the default value should be set as per the environments where the above environment needs to be set by users such as EC2, therefore using `localhost` makes more sense to me.
2. Since ports `4317` and `4318` are used by OTel, [for AppSignals we use](https://github.com/aws/amazon-cloudwatch-agent-operator/blame/0bb89bd770fa2ff4fc74979110a11db8bfffbaa9/internal/manifests/collector/ports.go#L35) `4315` (grpc) and `4316` (http).

So the default address `http://localhost:4315` is for users running CWAgent for AppSignals but maybe forgot to manually set the `OTEL_AWS_SMP_EXPORTER_ENDPOINT` config on their application.

Reply:
Hm I see the argument. This aligns with [public documentation](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Application-Signals-Enable-EC2.html) which tells customers to manually set `OTEL_AWS_SMP_EXPORTER_ENDPOINT` to `http://localhost:4315` ([ECS instructions](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Application-Signals-Enable-ECS.html) set to 127.0.0.1, which should resolve to the same thing?) while EKS instructions [[1](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Application-Signals-Enable-EKS-sample.html)][[2](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Application-Signals-Enable-EKS-Console.html)] does not reference this variable. Seems reasonable to me.